### PR TITLE
doc: add udgrade from v3.4 to v3.5

### PR DESCRIPTION
**Pull request by humblec**
This commit adds the upgrade documentation from v3.4 to v3.5

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: docs/ceph-csi-upgrade.md [hunk 1]
```diff
@@ -65,6 +66,38 @@ controller and snapshot CRD. more info can be found
...
+```console
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+```
+
+To upgrade CephFS refer section [upgrade cephfs](#upgrading-cephfs) and
+to upgrade RBD refer section [upgrade rbd](#upgrading-rbd).
+
+With that, we have successfully upgraded cephfs csi from v3.4 to v3.5
+
 ## Upgrading from v3.3 to v3.4
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md]: we don't need a separate section for 3.3 to 3.4 updates. just add a link how we are doing for 3.2 to 3.3 release https://github.com/ceph/ceph-csi/blob/9370d9d8bc777e4a59afcf8f544a053267f50ceb/docs/ceph-csi-upgrade.md#upgrading-from-v32-to-v33

File: docs/ceph-csi-upgrade.md [hunk 2]
```diff
@@ -65,6 +66,38 @@ controller and snapshot CRD. more info can be found
...
+
+git checkout v3.5.0 tag
+
+```bash
+git clone https://github.com/ceph/ceph-csi.git
+cd ./ceph-csi
+git checkout v3.5.0
+```
+
+**Note:** While upgrading please Ignore warning messages from kubectl output
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md]: change `we have successfully upgraded cephfs csi from v3.3 to v3.4` to 3.5 in multiple places

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: docs/ceph-csi-upgrade.md [hunk 3]
```diff
@@ -75,19 +81,24 @@ that will not be supported in the official releases. Builds from the devel
...
 cd ./ceph-csi
-git checkout v3.4.0
+git checkout v3.5.0
 ```
 
 **Note:** While upgrading please Ignore warning messages from kubectl output
 
+```console
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+```
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md]: this is already done at line 102

File: docs/ceph-csi-upgrade.md [hunk 4]
```diff
@@ -208,7 +219,7 @@ For each node:
   - The pod deletion causes the pods to be restarted and updated automatically
     on the node.
 
-we have successfully upgraded cephfs csi from v3.3 to v3.4
+we have successfully upgraded cephfs csi from v3.4 to v3.5
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md:217]: same change is required for rbd block

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: docs/ceph-csi-upgrade.md [hunk 5]
```diff
@@ -92,6 +98,8 @@ git checkout v3.4.0
 Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
 ```
 
+**Note:** While upgrading please Ignore warning messages from kubectl output
```
Comment (Madhu-1) [inline docs/ceph-csi-upgrade.md]:
```suggestion
**Note:** While upgrading please ignore above warning message from kubectl output
```

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Comment by humblec** [comment]
@Mergifyio refresh

**Pull request merged**